app/main.py

```diff
@@ -2,5 +2,18 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_arguments: list = []
+    cache_results: list = []
```

Comment:
use dict to store cache

Reply:
Updated! I liked the solution with lists more, though, because that's going to be one ugly dictionary for human reading (in case we need such functionality one day).